pandas/core/internals/construction.py

```diff
@@ -743,8 +743,10 @@ def sanitize_index(data, index: Index):
     Sanitize an index type to return an ndarray of the underlying, pass
     through a non-Index.
     """
-    if len(data) != len(index):
-        raise ValueError("Length of values does not match length of index")
+    if len(data) > len(index):
```

Comment:
you can just embed the len in the error message is better

Reply:
it still would not tell you which had a greater length without significant change. I feel like the length is irrelevant if you know which is larger. I could be wrong though. Please continue if you feel I'm overlooking something.

Reply:
There is also the problem testing the length as your tests currently message match

Reply:
I agree with @jreback  `f"Length of values ({len(values)}) does not match length of index ({len(index)})"` would be a better error message.

The unit test that tests this would have to change (that's fine).